### PR TITLE
SMP: Iterative Authalic Parameterization

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

Hardik Jain (https://hrdkjain.github.io/) has expressed the wish to contribute the code of his publication: Jain, Hardik, Manuel Wollhaf, and Olaf Hellwich. "Learning to Reconstruct Symmetric Shapes using Planar Parameterization of 3D Surface." Proceedings of the IEEE International Conference on Computer Vision Workshops. 2019. into CGAL.

This parameterization method fits into the authalic (area-preserving) family and although providing no guarantees (and sometimes indeed producing non-one-to-one parameterization), it gives better results than our current methods. 

Along the way:
- Fix some hashing issue when using a `Seam_mesh`
- Add distortion measurement tools
- Enhance SMP's tests
- Better documentation of SMP in general (proper documentation of basic types for parameterizers, cleaner lists in the user manual, etc.)

* **TODOs**
   *  [x] @sloriot: Filter branch (current size is 1.2Mo)

## Release Management

* Affected package(s): `Surface_mesh_paremeterization`
* Issue(s) solved (if any): -
* Feature/Small Feature (if any): [Iterative Authalic Parameterization](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Iterative_authalic) -- pre-appove on 2020/10/13
* Link to compiled documentation (obligatory for small feature): [SMP User Manual](https://cgal.geometryfactory.com/~mrouxell/SMP-Iterative_authalic/Surface_mesh_parameterization/#title11) 
* License and copyright ownership: tranfser to GeometryFactory

### CODE REVIEW

**Comment by lrineau** [comment]
There is a link error, there:
https://travis-ci.com/github/CGAL/cgal/jobs/369724815#L3259

**Comment by afabri** [comment]
What is probably missing is a statement why  and/or under which circumstances this method is better.

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Iterative_authalic_parameterizer_3.h [hunk 1]
```diff
@@ -0,0 +1,1163 @@
...
+///         and must be a model of `Parameterizer_3`.<br>
+///         <b>%Default:</b>
+/// \code
+///   Circular_border_arc_length_parameterizer_3<TriangleMesh_>
+/// \endcode
+///
+/// \tparam SolverTraits_ must be a model of `SparseLinearAlgebraTraits_d`.<br>
+///         Note that the system is *not* symmetric because border vertices are not removed from the system.<br>
+///         <b>%Default:</b> If \ref thirdpartyEigen "Eigen" 3.1 (or greater) is available
+///         and `CGAL_EIGEN3_ENABLED` is defined, then an overload of `Eigen_solver_traits`
```
Comment (sloriot) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Iterative_authalic_parameterizer_3.h:80]:
```suggestion
///         and `CGAL_EIGEN3_ENABLED` is defined, then the following class
```

**Comment by maxGimeno** [comment]
Problems : https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-28.shtml#Surface_mesh_parameterization

**Comment by maxGimeno** [comment]
warnings : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-Ic-30/Surface_mesh_parameterization/TestReport_Friedrich_Ubuntu-gcc-7.gz

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-33.shtml.

**Comment by lrineau** [comment]
> **TODOs**
> 
>     * [ ]  @sloriot: Filter branch (current size is 1.2Mo)

Is there something to do? Maybe that branch size is the right one.

**Comment by lrineau** [comment]
> > **TODOs**
> > ```
> > * [ ]  @sloriot: Filter branch (current size is 1.2Mo)
> > ```
> 
> Is there something to do? Maybe that branch size is the right one.

ping @sloriot

**Comment by sloriot** [comment]
I got 1.9Mo which seems too much, I'm investigating...

**Comment by sloriot** [comment]
581K now. Diff is limited to figs so it is still safe to merge it from the last "tested" flag.

**Pull request merged**